File: src/main/java/io/github/liquidTruffle/parser/ast/nodes/LiquidObjectNode.java
package io.github.liquidTruffle.parser.ast.nodes;

import com.oracle.truffle.api.frame.VirtualFrame;
import com.oracle.truffle.api.nodes.NodeInfo;
import io.github.liquidTruffle.parser.ast.AstNode;

@NodeInfo(description = "Represents Liquid objects that hold a single child representing the entire filter chain")
public class LiquidObjectNode extends AstNode {
    @Child
    private AstNode child;  // Either a literal/variable or the root of a filter chain

    public LiquidObjectNode(AstNode child) {
        this.child = child;
    }

    @Override
    public Object executeGeneric(VirtualFrame frame) {
        return child.executeGeneric(frame);
    }

    // Expose child for testing purposes
    public AstNode getChild() {
        return this.child;
    }
}

File: src/main/kotlin/io/github/liquidTruffle/parser/ast/nodes/LiquidObjectNode.kt
package io.github.liquidTruffle.parser.ast.nodes

import io.github.liquidTruffle.parser.ast.AstNode
import com.oracle.truffle.api.frame.VirtualFrame
import com.oracle.truffle.api.nodes.NodeInfo

@NodeInfo(description = "Represents Liquid objects that hold a child object, which can be either a literal or a variable")
class LiquidObjectNode(
	private val child: AstNode
) : AstNode() {
	
	override fun executeGeneric(frame: VirtualFrame): Any? {
		return child.executeGeneric(frame)
	}
	
	// Expose child for testing purposes
	val childNode: AstNode get() = this.child
}